feature/amazon/src/test/java/com/revenuecat/purchases/amazon/PriceExtractorTest.kt

```diff
@@ -0,0 +1,390 @@
+package com.revenuecat.purchases.amazon
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.mockk.every
+import io.mockk.mockk
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.Currency
+
+@RunWith(AndroidJUnit4::class)
+class PriceExtractorTest {
+
+    private lateinit var mockCurrency: Currency
+
+    @Before
+    fun setup() {
+        mockCurrency = mockk()
+    }
+
+    @Test
+    fun `US marketplace 7 USD`() {
+        mockCurrency("$", "USD")
+
+        val (currencyCode, priceAmountMicros) = "$7.00".extractPrice(mockCurrency)
```

Comment:
All the tests are basically the same with different inputs and expected outputs, so they could be written with `@ParameterizedTest` to remove the duplicated code.

Reply:
We need JUnit 5 for that sadly... And for JUnit 5, we need to target Java 8 and we are not doing that at the moment because we haven't really prioritized it. I will make sure this gets done in Q1 since it has plenty of benefits.

Reply:
https://app.shortcut.com/revenuecat/story/12126/upgrade-to-junit-5-in-android-tests